Physician Login Functionality

raw-chicken opened the pull request:
## Relevant Issue
[PAL-32](https://pallinet.atlassian.net/browse/PAL-32)
[PAL-33](https://pallinet.atlassian.net/browse/PAL-33)

## Description
* Modified patient login functions to be generalized for any type of users
* Added validation for Physician login
* Added easy debugging methods

jlei46 reviewed (approved):
Both debugging and normal login works. However, it did break the physician profile page so that will have to be fixed sometime in the future.
![image](https://user-images.githubusercontent.com/91035006/218983219-3354d67a-2a9e-49d3-a7b8-add7e4d57d92.png)
![image](https://user-images.githubusercontent.com/91035006/218983243-26843317-2829-423f-afb6-173319e97951.png)

raw-chicken commented:
The physician model issue seems to be that along some point, the Physician data was changed inside Firestore to a plain string, whereas before it was:
```
name: {
   family: ...,
   given: ...,
   text: ...
}
```
I've fixed the physician retrieval so that it is in the above format and is retrieving the information correctly.

In addition, I've gone in and changed the `birthDate` values in the practitioner's list of patients to be `birthdate`.
